src/proto/codec.rs

```diff
@@ -142,16 +140,15 @@ impl Decoder for WebSocketProtocol {
                 ensure_buffer_has_space!(src, offset + 8);
                 // SAFETY: The ensure_buffer_has_space call has validated this
                 // A conversion from 8 u8s to a u64 cannot fail
-                payload_length = u64::from_be_bytes(unsafe {
-                    src.get_unchecked(2..10).try_into().unwrap_unchecked()
-                }) as usize;
+                payload_length =
+                    u64::from_be_bytes(unsafe { src.get_unchecked(2..10).try_into().unwrap() })
+                        as usize;
                 if u16::try_from(payload_length).is_ok() {
                     return Err(Error::Protocol(ProtocolError::InvalidPayloadLength));
                 }
                 offset = 10;
             } else {
-                // SAFETY: Constructed from 7 bits so the max value is 127
-                unsafe { unreachable_unchecked() }
+                debug_assert!(false, "7 bit value expected to be <= 127");
```

Comment:
This may be provable. Otherwise it should at least remove the `else` branch just like `unreachable_unchecked` did.

Reply:
It will remove the `else` branch. See https://godbolt.org/z/953YYWof6